addons/transitions/Transitions.gd

```diff
@@ -158,12 +158,11 @@ func _common_post_fade(data:Array, new_scene) -> void:
 func _set_scene(new_scene:Node):
 	# Dispose old scene so we don't get any camera jitters or wierdness.
 	var previous_scene = _get_current_scene() if (_current_scene == null) else _current_scene
-	
 	previous_scene.queue_free()
-	print("killed previous scene %s" % previous_scene)
-	
-	scene_container.call_deferred("add_child", new_scene)
+	print_debug("killed previous scene %s" % previous_scene)
 	
+	if new_scene.get_parent() != scene_container:
+		scene_container.call_deferred("add_child", new_scene)
```

Comment:
Wouldn't this throw an error that the new scene already has a parent node?

What is the use-case for this? I usually create a new instance of a scene, call some methods, and then call fade methods; so the new scene isn't in the tree yet.

Reply:
In my case, I add the scene before going for the transition for running initialization methods (e.g setting up buttons on the scene) before running the transition. Therefore visual bugs can be avoided, like buttons moving during the transition.